src/js/services/id.js

```diff
@@ -225,6 +225,21 @@ module.factory('rpId', ['$rootScope', '$location', '$route', '$routeParams', '$t
     store.set('ripple_auth', {url:url, username: username, keys: keys});
   };
 
+  Id.prototype.setDeviceID = function (username, device_id) {
+    Id.normalizeUsernameForInternals(username);
+    return store.set(username + '|device_id', device_id);  
+  };
```

Comment:
Are you setting the normalized username, or the passed in username on the store?